assets/app/scripts/directives/fromFile.js

```diff
@@ -236,9 +236,15 @@ angular.module("openshiftConsole")
           return DataService.get(resourceGroupVersion, item.metadata.name, $scope.context, {errorNotification: false}).then(
             // resource does exist
             function(resource) {
-              if (item.kind !== "Template") {
+              // For Template update it's Objects, Paramaters and Labels fields. For other resources update their Spec.
+              if (item.kind === "Template") {
+                resource.objects = item.objects;
+                resource.parameters = item.parameters;
+                resource.labels = item.labels;
```

Comment:
I'm hesitant to hard-code the fields to update if we can avoid it. It's too easy to miss something, or something is added later and we lose data.

@fabianofranz what is the CLI logic for `oc replace`?